third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -74,6 +74,11 @@ namespace {
 // 1. The user provides a `num_stages` that specifies how many stages the
 //    pipeline will have. The number of stages must be larger than the distance
 //    from the first independent load to the compute in order to pipeline.
+//    1.a. User may also specify `global_prefetch = <n>` which will add <n>
```

Comment:
Hmm, I feel this is in a bit strange state. I can understand the desire to be able to experiment different schedules. This opens more experimention space, but it's not enough to express even common possibilities. the decision to only allow explicitly control prefetching global/local distance feels arbitary--why not allow tweaking other stages? If the goal is to have a mechanism for trying out different schedules, maybe we should allow specifying the exact stages for each isa category.

Reply:
I have updated to allow specify GLOBAL_LOAD stage, LOCAL_STORE stage, LOCAL_LOAD stage, COMPUTE stage. But in reality GLOBAL_LOAD must be first stage and COMPUTE must be last stage. So we only need to specify LOCAL_STORE and LOCAL_LOAD stage, which aligns with `global_prefetch` and `local_prefetch`.